### PR TITLE
CI Action: Build Image on PR label

### PR DESCRIPTION
**Pull request by irq0**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

### CODE REVIEW

**Comment by irq0** [comment]
Killed "Test s3gw", as it doesn't test anything related to this change

File: .github/workflows/build-image-on-pr-label.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    types: [ labeled ]
+
+  workflow_dispatch:
+
+env:
+  # Note: max 128 chars according to OCI Distribution Spec
+  IMAGE_TAG: pr-${{ github.workflow_sha }}-${{ github.run_id }}-${{ github.run_attempt }}
+
+jobs:
+  tests:
```
Comment (m-ildefons) [inline .github/workflows/build-image-on-pr-label.yaml:17]: Nit: rename to "build"

File: .github/workflows/build-image-on-pr-label.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          repository: aquarist-labs/s3gw
+          path: s3gw
+          submodules: false
+
+      - name: Checkout ceph
+        uses: actions/checkout@v3
+        with:
+          path: s3gw/ceph
+          submodules: recursive
+
```
Comment (m-ildefons) [inline .github/workflows/build-image-on-pr-label.yaml:33]: Nit: double empty line

File: .github/workflows/build-image-on-pr-label.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
```
Comment (m-ildefons) [inline .github/workflows/build-image-on-pr-label.yaml:38]: Is this needed anywhere?

File: .github/workflows/build-image-on-pr-label.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  workflow_dispatch:
+
+env:
+  # Note: max 128 chars according to OCI Distribution Spec
+  IMAGE_TAG: pr-${{ github.workflow_sha }}-${{ github.run_id }}-${{ github.run_attempt }}
+
+jobs:
+  tests:
+    if: github.event.label.name == 'ci/build-s3gw-image' || github.event_name == 'workflow_dispatch'
```
Comment (m-ildefons) [inline .github/workflows/build-image-on-pr-label.yaml:18]: Nit: I'd choose a shorter label for convenience: `ci/build`

**Review by m-ildefons — APPROVED** [review]
LPGTM, approved. Found some nits, no blockers though.

**Pull request merged**